Core/include/Acts/Propagator/RiddersPropagator.ipp

```diff
@@ -17,42 +17,31 @@ auto Acts::RiddersPropagator<propagator_t>::propagate(
       action_list_t_result_t<CurvilinearTrackParameters,
                              typename propagator_options_t::action_list_type>>;
 
+  auto startWithoutCov = start;
+  startWithoutCov.covariance() = std::nullopt;
+
   // Propagate the nominal parameters
-  auto nominalRet = m_propagator.propagate(start, options);
-  if (not nominalRet.ok()) {
-    return ThisResult::failure(nominalRet.error());
+  auto result = m_propagator.propagate(startWithoutCov, options);
+  if (not result.ok()) {
+    return ThisResult::failure(result.error());
   }
-
   // Extract results from the nominal propagation
-  auto nominalResult = std::move(nominalRet).value();
-  const BoundVector& nominalParameters =
-      nominalResult.endParameters->parameters();
-  // Use the curvilinear surface of the propagated parameters as target
-  const Surface& surface = nominalResult.endParameters->referenceSurface();
+  auto nominalResult = result.value();
+  assert(nominalResult.endParameters);
+  const auto& nominalFinalParameters = *nominalResult.endParameters;
 
-  // Steps for estimating derivatives
-  std::vector<double> deviations = {-4e-4, -2e-4, 2e-4, 4e-4};
-
-  // Allow larger distances for the oscillation
-  propagator_options_t opts = options;
-  opts.pathLimit *= 2.;
-
-  // Derivations of each parameter around the nominal parameters
-  std::array<std::vector<BoundVector>, eBoundSize> derivatives;
+  auto jacobian = wiggleAndCalculateJacobian(options, start, nominalResult);
+  nominalResult.transportJacobian = jacobian;
 
-  // Wiggle each dimension individually
-  for (unsigned int i = 0; i < eBoundSize; i++) {
-    derivatives[i] =
-        wiggleDimension(opts, start, i, surface, nominalParameters, deviations);
-  }
   if (start.covariance()) {
-    auto cov =
-        calculateCovariance(derivatives, *start.covariance(), deviations);
+    // use nominal parameters and Ridders covariance
+    auto cov = jacobian * (*start.covariance()) * jacobian.transpose();
```

Comment:
Ok I see, you refactored the function to only do the jacobian, and then the covariance can be calculated *inline* almost trivially. :+1: